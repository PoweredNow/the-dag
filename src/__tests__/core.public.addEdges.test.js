const addEdges = require('../core/public/addEdges');
const addNodes = require('../core/public/addNodes');
const { getState } = require('../core/private/_state');
describe('core.public.addEdges', () => {
  test('exports', () => {
    expect(addEdges).toMatchSnapshot();
  });
  test('addEdges throws if source and/or target are not registered nodes in state', () => {
    try {
      addEdges([{ source: 1, target: 2 }]);
    } catch (err) {
      expect(err.message).toMatchSnapshot();
    }
  });
  test('addEdges works when source and target exist in state', () => {
    addNodes([{ nodeID: 1 }, { nodeID: 2 }]);
    addEdges([{ source: { nodeID: 1 }, target: { nodeID: 2 } }]);
    const state = getState();
    expect(state).toMatchSnapshot();
  });

  test('Fails silently with no arguments', () => {
    addEdges();
    const state = getState();
    expect(state).toMatchSnapshot();
  });

  test('Returns silently when an edge already exists', () => {
    addNodes([{ nodeID: 1 }, { nodeID: 2 }]);
    addEdges([{ source: { nodeID: 1 }, target: { nodeID: 2 } }]);
    addEdges([{ source: { nodeID: 1 }, target: { nodeID: 2 } }]);
    const state = getState();
    expect(state).toMatchSnapshot();
  });
});
