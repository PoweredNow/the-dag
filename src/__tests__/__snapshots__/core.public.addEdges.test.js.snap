// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.public.addEdges Fails silently with no arguments 1`] = `
Object {
  "edges": Object {
    "1_2": Object {
      "edgeData": Object {},
      "edgeID": "1_2",
      "sourceID": 1,
      "targetID": 2,
    },
  },
  "nodes": Object {
    "1": Object {
      "nodeData": undefined,
      "nodeID": 1,
      "possibleTargets": Array [
        2,
      ],
    },
    "2": Object {
      "nodeData": undefined,
      "nodeID": 2,
      "possibleSources": Array [
        1,
      ],
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.addEdges Returns silently when an edge already exists 1`] = `
Object {
  "edges": Object {
    "1_2": Object {
      "edgeData": Object {},
      "edgeID": "1_2",
      "sourceID": 1,
      "targetID": 2,
    },
  },
  "nodes": Object {
    "1": Object {
      "nodeData": undefined,
      "nodeID": 1,
      "possibleTargets": Array [
        2,
      ],
    },
    "2": Object {
      "nodeData": undefined,
      "nodeID": 2,
      "possibleSources": Array [
        1,
      ],
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.addEdges addEdges throws if source and/or target are not registered nodes in state 1`] = `"Couldn't find source node 1"`;

exports[`core.public.addEdges addEdges works when source and target exist in state 1`] = `
Object {
  "edges": Object {
    "1_2": Object {
      "edgeData": Object {},
      "edgeID": "1_2",
      "sourceID": 1,
      "targetID": 2,
    },
  },
  "nodes": Object {
    "1": Object {
      "nodeData": undefined,
      "nodeID": 1,
      "possibleTargets": Array [
        2,
      ],
    },
    "2": Object {
      "nodeData": undefined,
      "nodeID": 2,
      "possibleSources": Array [
        1,
      ],
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.addEdges exports 1`] = `[Function]`;
